csharp-selenium-webdriver-sample/WebDriverFactory.cs

```diff
@@ -21,13 +21,29 @@ public static IWebDriver GetWebDriver(BrowserType browser)
             switch (browser)
             {
                 case BrowserType.Chrome:
-                    // Check if Chrome web driver environment variable already defined so that Chrome version and the webdriver are the same, otherwise use Selenium.WebDriver.ChromeDriver to install it                    
+                    // So now we need to specify the ChromeDriver directory path
+                    // We check first if the enivronemnt variable was set, if so we use the path value
+                    // Usually if you test your project on an azure pipleine, the selenium Chrome driver is pre-installed
+                    // The paths to the folder containing it can be obtained from the environment variables named ChromeWebDriver (Google Chrome).
+                    // So  it'd be compatible with the Chrome browser versions installed on the same agent.
+                    // if it's not set, we use Selenium.WebDriver.ChromeDriver nuget to install it in the current directory
+                    // chromedriver.exe does not appear in Solution Explorer, but it is copied to bin folder from package folder when the build process.
                     var chromeDriverDirectory = Environment.GetEnvironmentVariable("ChromeWebDriver") ?? Environment.CurrentDirectory;
+
+                    // Initializes a new ChromeDriver using the specified path to the directory containing ChromeDriver.exe.
                     return new ChromeDriver(chromeDriverDirectory);
 
                 case BrowserType.Firefox:
-                    // Check if Gecko web driver environment variable already defined so that Firefox version and the webdriver are the same, otherwise use Selenium.WebDriver.GeckoDriver to install it                    
+                    // So now we need to specify the FirefoxDriver directory path
```

Comment:
Most of this is identical to the Chrome case above; I think it can probably just reference the above, rather than repeating the giant comment block